make test-unit fails on ARM64 (Apple Silicon) with undefined reference to __aarch64_ldadd8_sync

**What happened**:
Running make test-unit on ARM64 machines (e.g. Apple Silicon via Podman) fails at the link stage with a wall of linker errors:

```
/usr/bin/gcc: running gcc failed: exit status 1
...undefined reference to `__aarch64_ldadd8_sync'
...undefined reference to `__aarch64_swp1_sync'
...undefined reference to `__aarch64_ldadd4_sync'
collect2: error: ld returned 1 exit status
FAIL
make: *** [test-unit-epp] Error 1
```

**Root Cause**

The builder image (ghcr.io/llm-d/llm-d-builder:dev) is based on quay.io/projectquay/golang:1.25 which runs RHEL 8.7 with GCC 8.5. Go's -race flag requires the ThreadSanitizer (TSan) runtime, which on ARM64 uses LSE (Large System Extension) atomic emulation helpers (__aarch64_ldadd8_sync, __aarch64_swp1_sync, etc.). These helpers were added to GCC's libgcc in GCC 9.4 — GCC 8.5 does not include them.

The failure is reproducible on any ARM64 (Apple Silicon Mac).

**What you expected to happen**:
Successful tests execution

**How to reproduce it (as minimally and precisely as possible)**:
Run `make test-unit` on an ARM64 platform with Podman. (Docker will probably fail too, but I'm not sure.)


**Affected platforms**
- ARM64 only  (with Podman ?)
- x86_64 is unaffected

--- a/Dockerfile.builder
+++ b/Dockerfile.builder
@@ -11,7 +11,14 @@ ARG KUSTOMIZE_VERSION=v5.6.0
 ARG DOCKER_VERSION=29.3.0
 ARG DOCKER_BUILDX_VERSION=v0.32.1
 
-RUN dnf install -y podman && dnf clean all
+RUN dnf install -y podman gcc-toolset-12 && dnf clean all
+
+# The base image ships GCC 8 (RHEL 8 default), which lacks the ARM64 LSE
+# atomic emulation helpers (__aarch64_ldadd8_sync etc.) required by Go's
+# race detector. gcc-toolset-12 provides GCC 12 which includes them.
+# TODO: remove gcc-toolset-12 once this base image is updated to RHEL 9
+#       (GCC 11+ ships natively there and includes the helpers).
+ENV PATH=/opt/rh/gcc-toolset-12/root/usr/bin:$PATH
 
 # Install docker CLI and buildx plugin
 RUN ARCH=$(uname -m) && \

